### DISCUSSION TITLE
NativeAoT support ?

### DISCUSSION THREAD

**Discussion by dax-leo** [alirezanet/Gridify#104] [General]
Hi,

great work with this library!
I was wondering if Gridify support NativeAoT ?

Thnx
Dax

**Comment by alirezanet** [comment]
Hi @dax-leo, 
good question :), I never tried to compile it for NativeAoT before, I think it must be straightforward since this library doesn't have any third-party dependencies. I'm planning to check it out later...

**Comment by dax-leo** [comment]
Just in case someone wants to play with NativAoT, I did quick investigation. 
Maybe someone smarter can find out more.

Filtering won't work without field mapping (ok, can live with that):

![image](https://github.com/alirezanet/Gridify/assets/70173560/552c2e6c-2963-4d79-b07b-07c38cc01f40)

OrderBy is also causing problems (expected considering NativeAoT has problem digesting Expressions).
To make order expression work had to modify OrderByMember<T>:
![image](https://github.com/alirezanet/Gridify/assets/70173560/dbf1f0a2-39de-4057-97dc-ce241b29807c)
Otherwise it died somewhere in the CreateQuery() expression call :(

But with these modifications (for a simple query) result is produced without any error:
![image](https://github.com/alirezanet/Gridify/assets/70173560/626dc9b0-8b8f-4b8c-98c5-14536ccc3e16)

I also have to mention there is a way to help AoT compiler by instructing it using RD.xml file. Good luck to anyone who figures out how to do it.
